### ISSUE TITLE
Catch common typos on scope methods

### ISSUE THREAD

**Issue by OverZealous** [Gillespie59/eslint-plugin-angular#352]
One of the more annoying typos that comes up is calling `scope.watch` instead of `scope.$watch`.

It would be nice to disallow calling any of the common scope methods on `scope`, `$scope`, or `$rootScope` without the appropriate `$` prefix.

Only on these objects:
- `$scope`
- `scope`
- `$rootScope`

These method names are invalid:
- `new`
- `watch`
- `watchGroup`
- `watchCollection`
- `digest`
- `destroy`
- `eval`
- `evalAsync`
- `apply`
- `applyAsync`
- `on`
- `emit`
- `broadcast`

Similarly, it would be nice to error if someone types in `scope.$on('destroy', ...)` instead of `scope.$on('$destroy', ...)`.

**Comment by remcohaszing** [comment]
Actually, assigning custom methods onto `$scope` or `$rootScope` is probably always a bad idea. No matter if it's called `watch`, `$witch` or `witch`. I think this is already handled by the [controller-as](https://github.com/Gillespie59/eslint-plugin-angular/blob/master/docs/controller-as.md) rule.

If any of the typos described exists, I think this should be caught by a unittest, not a linter.

Also, I think this (silly) example is still valid:

```js
/**
 * Bind a handler for the event of the user watching this element.
 */
.directive('watch', function() {
  return {
    scope: {
      watch: '&'
    },
    link: function link(scope, element) {
      element.on('watch', function(event) {
        scope.on(event);
      });
    }
  };
})
```

**Comment by OverZealous** [comment]
@remcohaszing That response doesn't make any sense to me.

This isn't about assigning, this is about catching common typos on scope methods.  For example, it's easy to accidentally type `scope.on('someEvent')` rather than `scope.$on('someEvent')`.  This is just an annoyance, and sounds like the specific use case for a linter.  It's nice to catch the bug early on.

The `controller-as` method both is too restrictive (simple directives don't always need a controller), and solving the wrong issue.

I don't see how unit tests come into play at all; that argument could pretty much be used for anything a linter does.
